### PR TITLE
[SALEOR-4715] Add webhook events for metadata updates

### PR DESCRIPTION
**Pull request by rafalp**
This PR adds metadata updated events to webhooks and subscriptions:

- `GIFT_CARD_METADATA_UPDATED`
- `ORDER_METADATA_UPDATED`
- `FULFILLMENT_METADATA_UPDATED`
- `CUSTOMER_METADATA_UPDATED`
- `COLLECTION_METADATA_UPDATED`
- `PRODUCT_METADATA_UPDATED`
- `PRODUCT_VARIANT_METADATA_UPDATED`
- `CHECKOUT_METADATA_UPDATED`
- `SHIPPING_ZONE_METADATA_UPDATED`
- `WAREHOUSE_METADATA_UPDATED`
- `VOUCHER_METADATA_UPDATED`
- `TRANSACTION_ITEM_METADATA_UPDATED`

As per request, webhook payload for those includes only `id` field with global ID of an object which's metadata was updated and `meta` key containing event's metadata. In case of subscriptions, subscribed type implements only an `id` field.

# TODO

* [x] DRY plugin manager methods
* [x] DRY subscription types
* [x] Check tests and test subscriptions and webhooks
* [x] Rebase
* [x] Fix "Added in Saleor 3.8" comments on new Schema items.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4715 Changing the order metadata does not trigger the webhooks](https://app.clickup.com/t/2549495/SALEOR-4715)

**Review by jakubkuc — APPROVED** [review]
Please fix conflicts. Otherwise lgtm :)

**Comment by korycins** [comment]
@rafalp  Not sure if you already handled this, but keep in mind that docs, also need to be updated. For example section like this: https://docs.saleor.io/docs/3.x/developer/extending/apps/asynchronous-webhooks#available-webhook-events

File: saleor/plugins/webhook/plugin.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from ...core.taxes import TaxData, TaxType
 from ...core.utils.json_serializer import CustomJsonEncoder
 from ...payment import PaymentError, TransactionKind
-from ...payment.models import Payment
+from ...payment.models import Payment, TransactionItem
```
Comment (korycins) [inline saleor/plugins/webhook/plugin.py:14]: Would be worth to add tests for logic added to this file.  (tests for `_trigger_metadata_updated_event`, and for each new handler added to WebhookPlugin).

File: saleor/plugins/webhook/plugin.py [hunk 2]
```diff
@@ -358,6 +368,15 @@ def gift_card_deleted(self, gift_card: "GiftCard", previous_value: None) -> None
             WebhookEventAsyncType.GIFT_CARD_DELETED, gift_card
         )
 
+    def gift_card_metadata_updated(
```
Comment (korycins) [inline saleor/plugins/webhook/plugin.py:371]: Each method added to any plugin that is used by PluginManager, should be also added to `BasePlugin`.

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/webhook/plugin.py (hunk 1 above)
Comment (rafalp) [reply saleor/plugins/webhook/plugin.py:14]: I've added tests for plugin manager methods, but I'm unsure about `_trigger_metadata_updated_event` - I would have to test it directly on webhook's plugin instance, or split into util function.

**Review by korycins — APPROVED** [review]

**Pull request merged**